inst/htmlwidgets/lib/rstudio_leaflet/rstudio_leaflet.css

```
@@ -39,3 +39,9 @@ Fix for https://github.com/rstudio/rmarkdown/issues/1949 */
 	max-width: none !important;
 	max-height: none !important;
 }
+
+/* Add explicit alignment rule to avoid CSS wider CSS applying to legend content
+Fix for https://github.com/quarto-dev/quarto-cli/issues/7843 */
+.leaflet .legend.leaflet-control {
+  text-align: left;
+}
```

I think `.leaflet .legend` should be sufficient (or maybe `.leaflet-control` would be better?). If we take this change, we'd also want to use `start` instead of `left` so that we respect writing direction.

```suggestion
.leaflet .legend {
  text-align: start;
}
```